### PR TITLE
Favorite quizzes and questions

### PR DESCRIPTION
**Pull request by set120120**
Adding funtionality to favorite questions and quizzes for backend.

### CODE REVIEW

File: doc/api-design.yml [hunk 1]
```diff
@@ -444,6 +444,68 @@ components:
...
+          format: date-time
+          example: "2024-11-23T12:45:30"
+        updatedAt:
+          type: string
+          format: date-time
+          example: "2024-11-23T12:45:30"
+    FavoriteQuestionRequest:
+      type: object
+      properties:
+        userId:
```
Comment (semihYILMAZ37) [inline doc/api-design.yml:479]: this should not include userId

File: doc/api-design.yml [hunk 2]
```diff
@@ -444,6 +444,68 @@ components:
           type: string
           format: date-time
           description: Timestamp when the reply was last updated.
+    FavoriteQuizRequest:
+      type: object
+      properties:
+        userId:
```
Comment (semihYILMAZ37) [inline doc/api-design.yml:450]: this should not include userId

File: backend/src/main/java/com/quizzard/quizzard/model/request/FavoriteQuestionRequest.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.quizzard.quizzard.model.request;
+
+
+import lombok.Data;
+
+@Data
+public class FavoriteQuestionRequest {
+
+    private Long userId;
```
Comment (semihYILMAZ37) [inline backend/src/main/java/com/quizzard/quizzard/model/request/FavoriteQuestionRequest.java:9]: userId should be removed

File: backend/src/main/java/com/quizzard/quizzard/model/request/FavoriteQuizRequest.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package com.quizzard.quizzard.model.request;
+
+import lombok.Data;
+
+@Data
+public class FavoriteQuizRequest {
+
+    private Long userId;
```
Comment (semihYILMAZ37) [inline backend/src/main/java/com/quizzard/quizzard/model/request/FavoriteQuizRequest.java:8]: userId should be removed

**Review by semihYILMAZ37 — APPROVED** [review]
overall it looks ok. However it has a unused and confusing request body field which should be removed from the request object and API doc. thanks for the work.

**Comment by set120120** [comment]
I got it overall. Due to time constraints, I will make these changes later. Thanks for your review @semihYILMAZ37

**Pull request merged**